kubernetes/config/kube_config_yaml.c

```diff
@@ -430,25 +430,39 @@ int kubeyaml_load_kubeconfig(kubeconfig_t * kubeconfig)
 {
     static char fname[] = "kubeyaml_load_kubeconfig()";
 
-    /* Set a file input. */
+    if (kubeconfig->fileName && kubeconfig->buffer) {
+        fprintf(stderr, "%s: Cannot use both kubeconfig->fileName and kubeconfig->buffer.\n", fname);
+        return -1;
+    }
+
+    /* Set a file input or use the provided buffer. */
     FILE *input = NULL;
     if (kubeconfig->fileName) {
         input = fopen(kubeconfig->fileName, "rb");
         if (!input) {
             fprintf(stderr, "%s: Cannot open the file %s.[%s]\n", fname, kubeconfig->fileName, strerror(errno));
             return -1;
         }
-    } else {
-        fprintf(stderr, "%s: The kubeconf file name needs be set by kubeconfig->fileName .\n", fname);
+    }
+    else if (kubeconfig->buffer) {
```

Comment:
If this condition is not currently in use, delete it.

Reply:
This is needed in order to allow code to continue if `kubeconfig->buffer` is set. I did it like this in order to avoid code duplication on parser and document variable initialization

Reply:
Got it. Thank you for clarification.